### ISSUE TITLE
Slowin Down (Google) Sites

### ISSUE THREAD

**Issue by luziusalbin** [jrie/flagCookies#9]
had to deactivate the plugin, after it was causing very heavy slowdowns in google photos and youtube. any chance to get that working soon?

**Comment by jrie** [comment]
Hello @luziusalbin - there are, unfortunately, a lot of reviews/reports of slowdowns in the Chrome extension store of FlagCookies . I suppose you use Chrome as browser?

In Firefox I, at present, cannot reproduce this issue of slowdowns or high CPU usage at other websites, all seems to be okay as is, also after recent updates to the addon/extension.

Do you experience slowdowns only at this particular websites, or also at other websites? And, do these websites set/add a lot of cookies or cookie data?

I gladly would like to nail this down..

**Comment by luziusalbin** [comment]
Hi there!

Yes i use the newest version of chrome and it's slowing down google-services. i use google photos a lot and it's not usable anymore with this extension. loading is slowing down the whole browser until it's completely stuck. as well as youtube is loading at very low speed and the videos are alwas in minimum quality shown...

i cannot find any extreme CPU usage, it's just "blocking" something...

thank you!

**Comment by MrMxyzptlk** [comment]
It's the same for me; Chrome is unusable with Flag Cookies and any Google Sites.

**Comment by zwometer** [comment]
Chrome was super slow, sometimes completely unresponsive for me. Especially on Google Maps. I had to deactivate Flag Cookies. 

The "Utility: Network Service" was hogging CPU and Memory in the Chrome Task Manager when these slow-downs happen. 
I hope that helps to get it sorted quickly :)

**Comment by cheehieu** [comment]
Same issue for me using the Chrome browser. I mostly experienced it with Gmail and Google Calendar. Patiently waiting for a fix! 🥲

**Comment by cheehieu** [comment]
Same issue for me using Chrome on macOS. I mostly experienced the slowdown with Gmail and Google Calendar. Patiently waiting for a fix! 🥲

**Comment by jrie** [comment]
I currently do not have much available to spend time working on a possible solution because of personal matters.
But I appreciate your feedback in this matter. Because there seem to be a quite amount of people affected by this.

If you have some spare time, please take an eye on in case this behavior occurs at other websites too - which are not related to Google. For example pages with lots of cookie information which might change frequently. Which I presume make the largest noises in regards of "cookie data changes" and might produce some heavier loading times.

Anyhow, thank you all for your insights! :)

**Comment by jrie** [comment]
A commit 81e3fd1a2e4e15de082e65de782bf31e3ce3f5c7 , which might improve the situation is added. I could not exactly clarify if this was the culprit - but this fixes some error(s) thrown in Chrome.

Can someone of you guys retest with the current Github version of the addon? If you need some assistance on how to test, please let me know. Bare in mind, the manifest file `manifest_v3_chrome_browsers.json` has to be renamed to `manifest.json` (rename or delete the existing `manifest.json` file which is created for Firefox).

**Comment by ColinFarL** [comment]
@jrie I tested the latest version from `master` in my Chrome and unfortunately it still tanked the browser speed (notably Maps and YouTube) as much as before until I disabled it.

**Comment by KonoVitoDa** [comment]
I use Brave Browser (Chromium-based). For me it's slowing down Outlook, making it unusable; and also a few other sites, like Shopee, AliExpress and others I can't record at the moment.

**Comment by tom12011** [comment]
I wondered why my Brave browser had been acting up over the last couple of months and had just found out it was because of this extension. Curious its not happening on Firefox, wonder if it has something to do with Google's DRM war

**Comment by farOverNinethousand** [comment]
I can confirm the slowdowns and they're not only happening on MS/Google websites but on any so the title of this ticket may be changed.
OS: Win 10
Chrome: Version 123.0.6312.86 (Offizieller Build) (64-Bit)

Here, it was slowing down Chrome to a point where it was nearly unusable.

**Comment by jrie** [comment]
I am not happy about this issue, and I tried to figure out on how to troubleshoot this in (Googles) or other Chrome based browsers and I did not come far trying to debug this at present.

One main issue in my case, I am using a slower mobile connection because of (temporary) provider reasons. And for me, the pages open basically "equally" quick or slow and I do not see a difference if the addon is active or not. Also I do not see harsh CPU usages, what also was reported in some case, so it is kinda hard to reproduce.

One more question: Does this performance impact also occur if only a very few tabs are open/active, or even if only one tab is opened?

**Comment by jrie** [comment]
I checked once more using Opera - on Google Maps there is indeed a very heavy load and the page is rendered slowly. But I do not see how the addon is interfering.

**Comment by farOverNinethousand** [comment]
I just uninstalled FlagCookies and tried to reproduce this issue which I was able to do:

1. I had about 10 tabs open.
2. With outlook.live.com (logged-in), I was able to perfectly reproduce the problem.
I loaded the page first, then re-installed FlagCookies and the problem happened immediately.
Without FlagCookies the website still looks to be a heavy one but it loads quite fast and there is no "lag.

With FlagCookies: Website is nearly unresponsive, loading icon never disappears, tab itself nearly frozen.
Uninstall FlagCookies and the issue will go away immediately in the still open tab.
I can't reproduce it with the Microsoft website when I'm not logged in so if you got a test Microsoft account it's worth trying otherwise I'd be open to send you one.

**Comment by KonoVitoDa** [comment]
> One more question: Does this performance impact also occur if only a very few tabs are open/active, or even if only one tab is opened?

Yes, it does. Opening Outlook as the only one tab still shows the same behavior.

**Comment by jrie** [comment]
Some changes have been done. One big routine is not triggering anymore on "loading" state and is commented out. This might possible reduce a lot of overhead. Can anyone test and confirm?

There were also other changes, but this was not really related to the loading of a tab or other resources. More guards and other fixes related to the UI, UI update and others.

**Comment by KonoVitoDa** [comment]
I just tried it (v3.6.0) but the slowdown on Outlook still persists.

**Comment by jrie** [comment]
@KonoVitoDa : I guess you did so, but did you use the version of the repository by loading as temporary extensions (with the correct manifest.json) and disabling the browser installed version when testing?

**Comment by KonoVitoDa** [comment]
> @KonoVitoDa : I guess you did so, but did you use the version of the repository by loading as temporary extensions (with the correct manifest.json) and disabling the browser installed version when testing?

I didn't. I don't know how to install it this way. :(

**Comment by KonoVitoDa** [comment]
Oh, my bad, I just found how to do it: I downloaded the repository and loaded the folder `through the Extensions Manager (Load unpacked). The slowdown on Outlook and Shopee seems to be gone on my quick test.

**Comment by jrie** [comment]
@KonoVitoDa - that is good to know. Did you ensure you use the correct `manifest` file?
The default when downloaded is used for Firefox and the browser might show an error when loading the addon.

In Brave you have to use `manifest_v3_chrome_browsers.json`, delete (or rename) the existing `manifest.json` and rename the `manifest_v3_chrome_browsers.json` to `manifest.json` - and load the extension once more.

**Comment by KonoVitoDa** [comment]
Oh, sorry, I had just used the default manifest.json. Doing it the right way the bug appears to be back. Sorry for giving you false hopes. :(

**Comment by jrie** [comment]
Just as a matter of interest, the development is going on. I am currently in the process of rewriting a lot of the code and also trying to minimize the amount of requests and data the addon is reading and writing of and to the browser.

Also, a lot of testing has to be done in advance, because a lot of redundant code will be merged, especially for the cookie addition and removal functions.

**Comment by soxhi8** [comment]
I am experiencing these slowdowns on Google Tabs on Firefox 130.0.1. It looks like it correlated with the rollout of the new Google Layout, but that can be pure chance.
I attached some performance profiles for the offending tabs.
[Firefox 2024-09-27 22.20 profile.json.gz](https://github.com/user-attachments/files/17169531/Firefox.2024-09-27.22.20.profile.json.gz)
[Firefox 2024-09-27 22.23 profile.json.gz](https://github.com/user-attachments/files/17169532/Firefox.2024-09-27.22.23.profile.json.gz)
[Firefox 2024-09-27 22.26 profile.json.gz](https://github.com/user-attachments/files/17169533/Firefox.2024-09-27.22.26.profile.json.gz)

**Comment by jrie** [comment]
For reference, issue #11 relates to this issue and is likely not only affecting a particular website only. 

As for further considerations I would direct to this comment: https://github.com/jrie/flagCookies/issues/11#issuecomment-2417348357 - I assume that this would improve things dramatically.

@soxhi8 - sorry for the late response and that I did not respond to your comment. I did not yet check the performance profiles and will try to do so as soon as possible.

**Comment by jrie** [comment]
@soxhi8 - I checked out your recorded profiles and it seems that the "storage retrieval" is indeed the culprit.

Unfortunately, at present, I did not yet found a way to lighten the data retrieval.
But I made some improvements for the type checking.

You are welcome to test the current state of the addon in this repository.

**Comment by Vanquish081** [comment]
Very slow loading on gmail too, very very slow.
